src/File.php

```diff
@@ -246,6 +249,24 @@ public function getFilePath() : string
         return $this->filePath;
     }
 
+    /**
+     * @param string[] $metaInfo
+     * @return File
+     */
+    public function setMetaInfo(array $metaInfo): self
+    {
+        $this->metaInfo = $metaInfo;
+        return $this;
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getMetaInfo(): array
```

Comment:
@alucic I was considering this more of a helper function, since you cannot overwrite meta-information about the file (like filename or offset). Otherwise users can be under the impression that you can actually overwrite this information by calling `setMetaInfo` ("old" meta information always takes priority in this PR). This method would actually return only the information that can be changed.

WDYT?